src/autofix/autofix.ts

```diff
@@ -0,0 +1,334 @@
+	namespace: string;
+	moduleName: string;
+	exportName?: string;
+	propertyAccess?: string;
+	position: Position;
+	node?: ts.Identifier | ts.PropertyAccessExpression | ts.ElementAccessExpression;
+}
+
+export interface DeprecatedApiAccessNode {
+	apiName: string;
+	position: Position;
+	node?: ts.CallExpression | ts.Identifier | ts.PropertyAccessExpression | ts.ElementAccessExpression;
+}
+
+type ImportRequests = Map<string, {
+	nodeInfos: (DeprecatedApiAccessNode | GlobalPropertyAccessNodeInfo)[];
+	identifier?: string;
+}>;
+
+// type ModuleDeclarationInfo = ExistingModuleDeclarationInfo | NewModuleDeclarationInfo;
+
+export interface ExistingModuleDeclarationInfo {
+	moduleDeclaration: ModuleDeclaration;
+	importRequests: ImportRequests;
+}
+
+export interface NewModuleDeclarationInfo {
+	declareCall: ts.CallExpression;
+	requireCalls: Map<string, ts.CallExpression[]>;
+	importRequests: ImportRequests;
+	endPos?: number;
+}
+
+function createCompilerHost(sourceFiles: SourceFiles): ts.CompilerHost {
+	return {
+		getSourceFile: (fileName) => sourceFiles.get(fileName),
+		writeFile: () => undefined,
+		getDefaultLibFileName: () => "lib.d.ts",
+		useCaseSensitiveFileNames: () => false,
+		getCanonicalFileName: (fileName) => fileName,
+		getCurrentDirectory: () => "",
+		getNewLine: () => "\n",
+		fileExists: (fileName): boolean => sourceFiles.has(fileName),
+		readFile: () => "",
+		directoryExists: () => true,
+		getDirectories: () => [],
+	};
+}
+
+const compilerOptions = {
+	checkJs: false,
+	allowJs: true,
+	skipLibCheck: true,
+
+	target: ts.ScriptTarget.ES2022,
+	module: ts.ModuleKind.ES2022,
+	isolatedModules: true,
+	sourceMap: true,
+	suppressOutputPathCheck: true,
+	noLib: true,
+	noResolve: true,
+	allowNonTsExtensions: true,
+};
+
+function createProgram(inputFileNames: string[], host: ts.CompilerHost): ts.Program {
+	return ts.createProgram(inputFileNames, compilerOptions, host);
+}
+
+// eslint-disable-next-line @typescript-eslint/require-await
+export default async function ({
+	rootDir: _unused1,
+	namespace: _unused2,
+	resources,
+	context,
+}: AutofixOptions): Promise<AutofixResult> {
+	const sourceFiles: SourceFiles = new Map();
+	const resourcePaths = [];
+	for (const [resourcePath, resource] of resources) {
+		const sourceFile = ts.createSourceFile(
+			resourcePath,
+			resource.content,
+			{
+				languageVersion: ts.ScriptTarget.ES2022,
+				jsDocParsingMode: ts.JSDocParsingMode.ParseNone,
+			}
+		);
+		sourceFiles.set(resourcePath, sourceFile);
+		resourcePaths.push(resourcePath);
+	}
+
+	const compilerHost = createCompilerHost(sourceFiles);
+	const program = createProgram(resourcePaths, compilerHost);
+
+	const checker = program.getTypeChecker();
+	const res: AutofixResult = new Map();
+	for (const [resourcePath, sourceFile] of sourceFiles) {
+		const newContent = applyFixes(checker, sourceFile, resourcePath, resources.get(resourcePath)!, context);
+		if (newContent) {
+			res.set(resourcePath, newContent);
+		}
```

Comment:
We should catch any error when trying to apply an autofix and skip the file in that case. I'm not sure about how this should be logged then (verbose/warning/error?), but we should definitely not halt the execution, as this prevents the user from seeing any finding or getting files fixed at all.